modules/flowable-cmmn-converter/src/main/java/org/flowable/cmmn/converter/CmmnDiExtensionXmlConverter.java

```diff
@@ -0,0 +1,83 @@
+/* Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.flowable.cmmn.converter;
+
+import javax.xml.stream.XMLStreamReader;
+
+import org.flowable.cmmn.converter.exception.XMLException;
+import org.flowable.cmmn.model.BaseElement;
+import org.flowable.cmmn.model.CmmnDiEdge;
+import org.flowable.cmmn.model.GraphicInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Tijs Rademakers
+ */
+public class CmmnDiExtensionXmlConverter extends BaseCmmnXmlConverter {
+    
+    protected static final Logger LOGGER = LoggerFactory.getLogger(CmmnDiExtensionXmlConverter.class);
+    
+    @Override
+    public String getXMLElementName() {
+        return CmmnXmlConstants.ELEMENT_DI_EXTENSION;
+    }
+    
+    @Override
+    public boolean hasChildElements() {
+        return false;
+    }
+
+    @Override
+    protected BaseElement convert(XMLStreamReader xtr, ConversionHelper conversionHelper) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.